pkg/apis/core/validation/validation.go

```diff
@@ -5941,6 +5942,57 @@ func validateWindowsSecurityContextOptions(windowsOptions *core.WindowsSecurityC
 	return allErrs
 }
 
+func validateWindowsHostProcessPod(podSpec *core.PodSpec, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	containerCount := 0
+	hostProcessContainerCount := 0
+
+	podHostProcess := false
+	if podSpec.SecurityContext != nil && podSpec.SecurityContext.WindowsOptions != nil && podSpec.SecurityContext.WindowsOptions.HostProcess != nil {
+		podHostProcess = *podSpec.SecurityContext.WindowsOptions.HostProcess
+	}
+
+	hostNetwork := false
+	if podSpec.SecurityContext != nil && podSpec.SecurityContext.HostNetwork == true {
+		hostNetwork = true
+	}
+
+	// Note: Pod-wide WindowsSecurityContextOptions are inherited by containers that do not explicitly set them.
+	for _, c := range append(podSpec.Containers, podSpec.InitContainers...) {
+		containerCount++
+		if c.SecurityContext != nil && c.SecurityContext.WindowsOptions != nil {
+			if c.SecurityContext.WindowsOptions != nil && *c.SecurityContext.WindowsOptions.HostProcess {
+				hostProcessContainerCount++
+			}
+		} else if podHostProcess {
+			hostProcessContainerCount++
+		}
+	}
+
+	if hostProcessContainerCount > 0 {
```

Comment:
I think that is the behavior we want. See https://github.com/kubernetes/kubernetes/pull/99576/files/760cbaf64d92a3b297337cffb70b29945cec71d0#r588565175

Thoughts?

Reply:
Hmm... I think it would be OK to schedule the Pod here. With other fields in WindowsOptions, if all containers set them to a different value that what was set in the Pod settings then what is specified in the Pod's WindowsOptions would be ignored.

For example if the Pod sets `PodSecurityContext.WindowsOptions.RunAsUsername=User1` but then all the containers set `SecurityContext.WindowsOptions.RunAsUsername=User2` then all of the Pods would run as 'User2`. 

Thoughts?

Reply:
I updated the validation logic to fail in this case and added a test case to the unit tests.

Reply:
The behavior where if `podHostProcess = true` then we should fail pod validation if any container sets `hostProcess = false` is described in the KEP under https://github.com/kubernetes/enhancements/tree/master/keps/sig-windows/1981-windows-privileged-container-support#windowssecuritycontextoptionshostprocess-flag 

If PodSecurityContext.WindowsSecurityContextOptions.HostProcess = false no containers may set Container.SecurityContext.WindowsSecurityContextOptions.HostProcess = true

(Sorry i missed this before)